packages/admin-ui-docs/content/filter/code.mdx

```diff
@@ -0,0 +1,305 @@
+          <FilterOptionRadio id="#2" label="Empty" />
+          <FilterOptionRadio id="#3" label="Half full" />
+        </FilterListbox>
+        <FilterFooter />
+      </FilterPopover>
+
+      // multiple selection filter
+      <FilterDisclosure state={multipleSelectState}>Example</FilterDisclosure>
+
+      <FilterPopover state={multipleSelectState}>
+        <FilterListbox>
+          <FilterOptionCheckbox id="#1" label="Full" />
+          <FilterOptionCheckbox id="#2" label="Empty" />
+          <FilterOptionCheckbox id="#3" label="Half full" />
+        </FilterListbox>
+        <FilterFooter />
+      </FilterPopover>
+    <FilterGroup />
+  )
+}
+```
+
+</Section>
+
+<Section hasExplanation>
+
+## State
+
+The state hooks `useFilterState` and `useFilterMultipleState` contain all business logic needed for
+controlling individual filters. The two hooks are different, but they share many props in common.
+
+The state hook `useFilterGroupState` handles the "clear all" action for the `FilterGroup` component.
+
+</Section>
+
+<Section>
+
+### useFilterState and useFilterMultipleState props
+
+| Name           | Type                                           | Description                                                                      | Required | Default  |
+| -------------- | ---------------------------------------------- | -------------------------------------------------------------------------------- | -------- | -------- |
+| fullList       | {id: string, label: string, value?: AnyObject} | Full option list that will be used for searching. Only use it if you need search.| 🚫       | -        |
+
+
+both filters return an object that extends the following type
+
+```ts
+interface GenericFilterState {
+  /**
+   * Function called when clear button on filter footer is clicked
+   */
+  onClear: () => void
+  /**
+   * Function called when apply button on filter footer is clicked
+   */
+  onChange: () => void
+  /**
+   * State that handles the popover menu
+   */
+  menu: MenuState
+}
+```
+
+With additional props for each type
+
+```ts
+export interface FilterOption<T> {
+  id: string
+  label: string
+  /**
+   * Object that may contain anything 
+   */
+  value?: T
+}
+
+interface FilterState<T> extends GenericFilterState {
+  /**
+   * State that handles rendered options and search
+   */
+  combobox: ComboboxState<FilterOption<T>>
+  /**
+   * Option that is currently applied
+   */
+  appliedItem?: FilterOption<T>
+  /**
+   * Setter for appliedItem
+   */
+  setAppliedItem: (option: FilterOption<T>) => void
+  /**
+   * Shorthand function for fetching any key from the attached object (value) on appliedItem
+   */
+  getFromApplied: (key: string) => any
+}
+
+interface FilterMultipleState<T> extends GenericFilterState {
+  /**
+   * State that handles rendered options and search
+   */
+  combobox: ComboboxMultipleState<FilterOption<T>>
+  /**
+   * Options that are currently applied
+   */
+  appliedItems: ItemList<T>
```

Comment:
Would it be relevant to showcase these types as well? My only concern would be the documentation getting too long unnecessarily